src/providers/web3ProviderWrapper.ts

```diff
@@ -22,51 +22,43 @@
   in accordance with implementation of smart contract interfaces of ERC725
 */
 
-import * as abi from 'web3-eth-abi';
-
-import { METHODS } from '../lib/constants';
 import { JsonRpc } from '../types/JsonRpc';
 import { Method } from '../types/Method';
-
-// @ts-ignore
-const web3abi = abi.default;
-let idCount = 1;
-
-export default class Web3Source {
-  public provider: any;
-
-  constructor(provider) {
+import { constructJSONRPC, decodeResult } from '../lib/provider-wrapper-utils';
+import { ProviderTypes } from '../types/provider';
+
+export class Web3ProviderWrapper {
+  type: ProviderTypes;
```

Comment:
Is this part of the code getting "picked" by the typedoc? Would it make sense to add TypeDoc here and in the functions of the class so it goes into the docs?

Reply:
Yes, everything in `src` is getting picked up. However Fabian wanted to keep the docs small, and since we already have the `providers.md` I was thinking that will be sufficient. I'm cool with documenting this too, it's not a big effort.

Reply:
I'll document this when I unify the way providers are injected, potentially in the next few PRs